src/iks003_incompressibleLinearElasticity.cpp

```diff
@@ -15,84 +15,90 @@
   }
 
   using GlobalIndex = typename LocalView::MultiIndex;
   void globalFlatIndices(std::vector<GlobalIndex> &globalIndices) const {
-    const auto &feDisp = this->localView().tree().child(_0, 0).finiteElement();
+    const auto &feDisp = localView_.tree().child(_0, 0).finiteElement();
     for (size_t i = 0; i < feDisp.size(); ++i) {
       for (int j = 0; j < Traits::worlddim; ++j) {
-        globalIndices.push_back(this->localView().index((this->localView().tree().child(_0, j).localIndex(i))));
+        globalIndices.push_back(localView_.index((localView_.tree().child(_0, j).localIndex(i))));
       }
     }
-    const auto &fePressure = this->localView().tree().child(_1).finiteElement();
+    const auto &fePressure = localView_.tree().child(_1).finiteElement();
     for (size_t i = 0; i < fePressure.size(); ++i) {
-      globalIndices.push_back(this->localView().index((this->localView().tree().child(_1).localIndex(i))));
+      globalIndices.push_back(localView_.index((localView_.tree().child(_1).localIndex(i))));
     }
   }
 
- private:
+  inline double calculateScalar(const FERequirementType &par) const { return calculateScalarImpl<double>(par); }
+
+  [[nodiscard]] constexpr int size() const { return localView_.size(); }
+  const Element &gridElement() { return localView_.element(); }
+  const LocalView &localView() const { return localView_; }
+  LocalView &localView() { return localView_; }
+
+ protected:
   template <class ScalarType>
-  [[nodiscard]] ScalarType calculateScalarImpl(const FERequirementType &par,
-                                               const Eigen::VectorX<ScalarType> &dx) const {
+  auto calculateScalarImpl(const FERequirementType &par, const std::optional<const Eigen::VectorX<ScalarType>> &dx
+                                                         = std::nullopt) const -> ScalarType {
     const auto &d      = par.getGlobalSolution(Ikarus::FESolutions::displacement);
     const auto &lambda = par.getParameter(Ikarus::FEParameter::loadfactor);
-    Eigen::VectorX<ScalarType> localDisp(this->localView().size());
+    Eigen::VectorX<ScalarType> localDisp(localView_.size());
     localDisp.setZero();
-    auto &displacementNode = this->localView().tree().child(_0, 0);
-    auto &pressureNode     = this->localView().tree().child(_1);
+    auto &displacementNode = localView_.tree().child(_0, 0);
+    auto &pressureNode     = localView_.tree().child(_1);
     const auto &feDisp     = displacementNode.finiteElement();
     const auto &fePressure = pressureNode.finiteElement();
     Eigen::Matrix<ScalarType, Traits::dimension, Eigen::Dynamic> disp;
     disp.setZero(Eigen::NoChange, feDisp.size());
     for (auto i = 0U; i < feDisp.size(); ++i)
       for (auto k2 = 0U; k2 < Traits::mydim; ++k2)
-        disp.col(i)(k2)
-            = dx[i * 2 + k2] + d[this->localView().index(this->localView().tree().child(_0, k2).localIndex(i))[0]];
+        disp.col(i)(k2) = dx.value()[i * Traits::mydim + k2]
```

Comment:
This line fails now if calculateScalar is called due to `bad_optional_access`. Thats why i indicated this is not a good idea. Thus, this has to be wrapped with `if(dx)` which by itself is ugly and leads to code duplication.

Reply:
Therefore,., I wanted to disuss if we want this or not. (In the end we should write a wrapper anyway and remove the code duplication in ikarus and here. For the elemnts that are using PowerFE this could be a good use case for aq basclass implementation